Automattic/wp-calypso
# Anchor.FM Gutenboarding passes along an episode id as well

#### Changes proposed in this Pull Request

* Anchor.FM Gutenboarding passes along an episode id as well

#### Testing instructions

* Visit  http://calypso.localhost:3000/new?&anchor_podcast=22b6608&anchor_episode=e324a06c-3148-43a4-85d8-34c0d8222138 in an incognito window
* Go through the gutenboarding process, creating a site and account
* Check the network tab when you finish, the episode id from the initial url will be passed to the wpcom site creation
![2020-12-04_15-57](https://user-images.githubusercontent.com/937354/101219686-fff9b700-3649-11eb-9695-ecc8f707817e.png)
* You can also add `console.log( { anchorFmEpisodeId, anchorFmPodcastId } );` to `use-step-navigation` if you'd like

#### Notes

* Is this another hex style ID or is it a guid with "-"?
  * Anchor asked us to allow any characters allowed in a URL.
* Abstract function `useAnchorParameter()` was used to cut down on the repetition between `useAnchorFmEpisodeId`, `useAnchorFmPodcastId`
* Each parameter has three names. What comes in the query string, what is stored in location state, and what is sent to wpcom.
  * Initial Query String: `?anchor_podcast=cafe1`
  * Location State: `anchorFmPodcastId`
  * Sent to WPCOM: `anchor_fm_podcast_id`.
  * If we can align some of these, the code will be simpler.  For example, changing the initial url to http://calypso.localhost:3000/new?anchorFmPodcastId=cafe1&anchorFmEpisodeId=f00d2 will allow `useAnchorParameter()` to be simplified.
* There is some other repetition in `use-step-navigation`.. Personally, I am fine with it as long as there are only 2 parameters, and would consider reworking if we have any more.
